### ISSUE TITLE
Avoid duplicates in document search when multi-geo is enabled.

### ISSUE THREAD

**Issue by lsavard** [microsoft-search/pnp-modern-search#2519]
**Version used**
4.5.4
Issue also present in 3.15.1

**Describe the bug**
Since we enabled multi-geo for SharePoint/OneDrive last week, all PnP Search  webparts have been showing duplicate search results. Each file is shown twice in the results. Behavior was normal before enable multi-geo. We have now 2 locations, CAN and NAM.

**To Reproduce**
Any search on a tenant with multi-geo enabled

**Expected behavior**
No duplicates

Thanks!

**Comment by wobba** [comment]
That's an API issue, not web part issue. And if these are People results, then it's expected.

**Comment by lsavard** [comment]
Hi!
This has been closed a bit quickly; I might have not described the issue correctly. 
This is really a bug in the PnP Search Webparts. Please note this is about file search, not people search.
If we search using any other mean, i.e. built-in SP Search, builtin webparts like highlighted content, Search Query Tool, or even another webpart using search api (lightningtool conductor), the results are fine and not duplicated.
Only in PnP Search Webpart are the results duplicated. Each and every PnP search webpart now show 100% duplicated files in the results.
I know it is hard to test because Microsoft won't let you enable multi-geo on small tenants. We ourselves only have multi-geo enabled on our production tenant.
I tried to upload screenshots but it does an error message. Please let me know if I can do anything else to help resolve the issue.

Thanks!

**Comment by wobba** [comment]
In the query tool, add `EnableMultiGeoSearch:true` to the `Appended Query Properties` field, and you should get the same result as what you see in the web parts. If you say it's a web part issue, then check the JSON response on the API call and state that each file is only coming once - which seems unlikely.

**Comment by lsavard** [comment]
Hi! I did some additional tests as per your suggestion, and the findings are very interesting.
First, setting the EnableMultiGeoSearch:true value or not does not change the results (still no duplicates). However, changing the value for TrimDuplicates does change everything ! It defaults to true in SearchQueryTool, and when I set it to false, I get the duplicates. I can also confirm by looking at the request payload that PnP Search WebPart sends the value false for TrimDuplicates. 

So my remaining question would be : can I set the parameter TrimDuplicates to true in the PnP Search WebPart ? I am pretty sure that would resolve the issue.

Thanks!

**Comment by wobba** [comment]
Yes you can as it's available from v4.7, but I advise against it as the feature for most cases remove too much. I recommend logging a support ticket as each geo should not return items from other geo's (except for profile URL's). Are duplicates from OneDrive only or all sites?

**Comment by lsavard** [comment]
Duplicates are from all sharepoint sites. I did not test it on OneDrives as we have no need for PnP search on OneDrives as of now.
I agree it does not feel normal each geos return items from all geos, but I doubt MS will change that even if we ask. As their own search interfaces works fine as it is.
Thanks for v4 I can see the toggle. Does the latest v3.x also support the TrimDuplicates toggle?

**Comment by lsavard** [comment]
Hi, with some additional testing, I can confirm that your advice is good. The TrimDuplicates options seems to remove too much results. It does not work correctly and will create more problems if we enable it.
I did a new test with EnableMultiGeoSearch:true and this time I could clearly see the number of results doubling (128=>256).
I also see in the payload sent by the browser that the PnPSearch webpart sends the property EnableMultiGeoSearch:true.
So, is it possible to have the WebPart to not send it, or send EnableMultiGeoSearch:false ?
That would be the right fix.
Thanks again !

**Comment by wobba** [comment]
@lsavard if you don't enable multi-geo results will come from the geo the web part is hosted on. If you have the search page on one geo but want another geo to show, then enable multigeo and add something like below to limit results to one particular GEO:

`GeoLocation:EUR` <- EUR geo.. replace with the value for your geo.

**Comment by lsavard** [comment]
Hello ! I am happy to report we have finally come to a resolution for the multi-geo search problem. I recompiled the sources for the 4.7 release to remove these lines from SharePointSearchDataSource.ts :
        }, {
            Name: "EnableMultiGeoSearch",
            Value: {
                BoolVal: true,
                QueryPropertyValueTypeIndex: 3
            }

Since the component have been deployed to our production multi-geo tenant, the PnP search results webpart now behaves correcty. We get no more duplicate results. Also, and this could be counter-intuitive, but the searches are still returning results from the 2 geos.

In fact, what we are experiencing in reality is NOT what is documented [https://learn.microsoft.com/en-us/microsoft-365/enterprise/configure-search-for-multi-geo?view=o365-worldwide](https://learn.microsoft.com/en-us/microsoft-365/enterprise/configure-search-for-multi-geo?view=o365-worldwide). That documentation seems plainly wrong :(

Thanks again for taking the time to help, please let me know if there is a way to get this fix into the official PnP-modern-search builds !

**Comment by wobba** [comment]
You are right and I really thought we had added this as an option to toggle. Guess not. And if you get multi-geo without this setting that is weird. I'll take a look at this. But adding my proposed KQL would also have worked, and no rebuilding :)
